src/Tizen.System.Session/Session/Session.cs

```diff
@@ -0,0 +1,509 @@
+
+        private readonly object _switchUserCompletionLock = new object();
+
+        /// <summary>
+        /// Event to be invoked when an existing subession user is successfully switched to an acvite state.
+        /// </summary>
+        /// <exception cref="ArgumentException">Session UID of this object is invalid</exception>
+        /// <exception cref="OutOfMemoryException">Out of memory</exception>
+        /// <exception cref="IOException">Internal error</exception>
+        /// <exception cref="UnauthorizedAccessException">Not permitted</exception>
+        /// <exception cref="NotSupportedException">Not supported</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public event EventHandler<SwitchUserCompletionEventArgs> SwitchUserCompletion
+        {
+            add
+            {
+                lock (_switchUserCompletionLock)
+                {
+                    if (_switchUserCompletionHandler == null)
+                        RegisterCallbackForEvent(SessionEventType.SwitchUserCompletion, _switchUserCompletionCB, OnSwitchUserCompletion);
+                    _switchUserCompletionHandler += value;
+                }
+            }
+            remove
+            {
+                lock (_switchUserCompletionLock)
+                {
+                    _switchUserCompletionHandler -= value;
+                    if (_switchUserCompletionHandler == null)
+                        UnregisterCallbackForEvent(SessionEventType.SwitchUserCompletion, _switchUserCompletionCB);
+                }
+            }
+        }
+
+        private void CheckError(SessionError ret, string msg)
+        {
+            if (ret == SessionError.None)
+                return;
+
+            Log.Error(SessionErrorFactory.LogTag, msg);
+            SessionErrorFactory.ThrowException(ret);
+        }
+
+        private void IntPtrToStringArray(IntPtr unmanagedArray, int size, out string[] managedArray)
+        {
+            managedArray = new string[size];
+
+            byte[] byteArray = new byte[MaxUserLength * size];
+            byte[] tmpArray = new byte[MaxUserLength];
+
+            Marshal.Copy(unmanagedArray, byteArray, 0, MaxUserLength * size);
+
+            for (int i = 0; i < size; i++)
+            {
+                Array.Copy(byteArray, MaxUserLength * i, tmpArray, 0, MaxUserLength);
+                managedArray[i] = Encoding.UTF8.GetString(tmpArray);
+            }
+        }
+
+        private void RegisterCallbackForEvent(SessionEventType eventType, Interop.Session.SubsessionEventCallback eventCallback,
+            EventDelegate delegateToSet)
+        {
+            eventCallback = new Interop.Session.SubsessionEventCallback(delegateToSet);
+            SessionError ret = Interop.Session.SubesssionRegisterEventCallback(SessionUID, eventType,
+                eventCallback, IntPtr.Zero);
+            CheckError(ret, $"Interop failed to register a callback for an event of type {eventType}");
+        }
```

Comment:
It's wrong
`eventCallback` is not ref argument

Reply:
Yes. Looks like it needs to be `ref`. Will fix.